### PR TITLE
Files component 

### PR DESCRIPTION
**Pull request by carmenvramos**

### CODE REVIEW

File: src/components/File.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-function File({ title }) {
-  return <p>{title}</p>;
+function File({ file }) {
```
Comment (mehtaphysical) [inline src/components/File.js]: if you only need title, just pass in the title

File: src/containers/markdown/Files.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  state = {
+    files: [],
+    selectedFile: 0
+  };
+
+  componentDidMount() {
+    this.setState({ files: getFiles(store.getState()) });
+    this.unsubscribe = store.subscribe(() => {
+      this.setState({ files: getFiles(store.getState()) });
+      console.log('state from store', store.getState());
```
Comment (MariahAdams) [inline src/containers/markdown/Files.js]: Remove all console logs from production code.

File: src/containers/markdown/Files.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  updateSelectedFile = ({ target }) => {
+    store.dispatch(updateSelectedFile(target.id));
+  };
+
+  render() {
+    const { files } = this.state;
+
+    const filesList = files.map(file => {
```
Comment (MariahAdams) [inline src/containers/markdown/Files.js]: I would extract this file list into a Files presentational component that renders the child File component.

**Comment by MariahAdams** [comment]
Missing additional tests for new reducers and selectors. Please add for approval.

**Comment by mehtaphysical** [comment]
There are no tests and this adds much more functionality than the card asks for. Also, some of the functionality is incomplete.

File: src/actions/document.js [hunk 4]
```diff
@@ -1,5 +1,17 @@
...
+  payload: markdown
+});
+
+export const UPDATE_FILES = 'UPDATE_FILES';
+export const updateFiles = file => ({
+  type: UPDATE_FILES,
+  payload: file
+});
+
+export const UPDATE_SELECTED_FILE = 'UPDATE_SELECTED_FILE';
```
Comment (mehtaphysical) [inline src/actions/document.js]: The card just asks to display a list of markdown files. this seems outside the scope of that.

File: src/actions/document.js [hunk 5]
```diff
@@ -1,5 +1,17 @@
 export const UPDATE_MARKDOWN = 'UPDATE_MARKDOWN';
 export const updateMarkdown = markdown => ({
   type: UPDATE_MARKDOWN,
-  payload: { markdown }
+  payload: markdown
+});
+
+export const UPDATE_FILES = 'UPDATE_FILES';
+export const updateFiles = file => ({
```
Comment (mehtaphysical) [inline src/actions/document.js]: the card asks to display markdown files. this seems to be about this card: https://trello.com/c/wt5ED67E/8-create-container-to-save-markdown-files-4pts

File: src/containers/markdown/Files.js (hunk 2 above)
Comment (mehtaphysical) [inline src/containers/markdown/Files.js]: remove this console.log

File: src/containers/markdown/Files.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import { updateSelectedFile } from '../../actions/document';
+
+export default class Files extends PureComponent {
+  state = {
+    files: [],
+    selectedFile: 0
+  };
+
+  componentDidMount() {
+    this.setState({ files: getFiles(store.getState()) });
```
Comment (mehtaphysical) [inline src/containers/markdown/Files.js]: line #16 is the same as this. Maybe have a function for this?

File: src/containers/markdown/Files.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    store.dispatch(updateSelectedFile(target.id));
+  };
+
+  render() {
+    const { files } = this.state;
+
+    const filesList = files.map(file => {
+      return <li key={file.id}>
+        <File file={file}/>
+        <button id={file.id} onClick={this.updateSelectedFile}>Edit</button>
```
Comment (mehtaphysical) [inline src/containers/markdown/Files.js]: This seems outside the scope. Does it even work?

File: src/reducers/document.js [hunk 8]
```diff
@@ -1,15 +1,48 @@
...
+      };
+    case UPDATE_FILES:
+      return { 
+        ...state,
+        files: [
+          ...state.files,
+          action.payload
+        ]
+      };
+    case UPDATE_SELECTED_FILE:
```
Comment (mehtaphysical) [inline src/reducers/document.js]: this seems outside the scope

File: src/reducers/document.js [hunk 9]
```diff
@@ -1,15 +1,48 @@
...
 
 export default function reducer(state = initialState, action) {
   switch(action.type) {
     case UPDATE_MARKDOWN:
-      return { ...action.payload };
+      return {
+        ...state,
+        markdown: action.payload
+      };
+    case UPDATE_FILES:
```
Comment (mehtaphysical) [inline src/reducers/document.js]: this seems outside the scope

File: src/selectors/document.js [hunk 10]
```diff
@@ -1,2 +1,4 @@
 export const getMarkdown = state => state.markdown;
 
+export const getFiles = state => state.files;
```
Comment (mehtaphysical) [inline src/selectors/document.js:3]: Also probably want a selector to `getFileTitles`

**Review by mehtaphysical — APPROVED** [review]
